examples/filter_before_line_number.rs

```diff
@@ -0,0 +1,26 @@
+use std::{
+    ffi::CStr,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+
+use libc::c_char;
+use tree_sitter::Node;
+
+static ROW_NUMBER: AtomicUsize = AtomicUsize::new(0);
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
```

Comment:
Clippy was arguing that this whole `fn` should be marked `unsafe` because of the `CStr::from_ptr()` dereferencing of a raw pointer inside it

But I don't understand well enough what the argument/implications are (this is only being consumed via FFI which is `unsafe`-aware, no?) so didn't want to blindly go with it

Reply:
I guess any function accepting C-pointers (and doing something with them) is kind of unsafe by definition, but on the other hand, who cares if it's essentially the entry point to the whole library. I don't really see a downside to marking it unsafe though. 🤷🏻